amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/bulk/BulkBaseService.java

```diff
@@ -0,0 +1,249 @@
+
+    protected ServiceRegistry serviceRegistry;
+    protected SearchService searchService;
+    protected TransactionService transactionService;
+    protected SearchMapper searchMapper;
+    protected BulkMonitor<T> bulkMonitor;
+
+    protected int threadCount;
+    protected int batchSize;
+    protected int itemsPerTransaction;
+    protected int maxItems;
+    protected int loggingInterval;
+
+    @Override
+    public void afterPropertiesSet() throws Exception
+    {
+        this.searchService = serviceRegistry.getSearchService();
+    }
+
+    /**
+     * Execute bulk operation on node based on the search query results
+     *
+     * @param nodeRef       node reference
+     * @param bulkOperation bulk operation
+     * @return bulk status
+     */
+    public T execute(NodeRef nodeRef, BulkOperation bulkOperation)
+    {
+        checkPermissions(nodeRef, bulkOperation);
+
+        ResultSet resultSet = getTotalItems(bulkOperation.searchQuery(), maxItems);
+        if (maxItems < resultSet.getNumberFound() || resultSet.hasMore())
+        {
+            throw new InvalidArgumentException("Too many items to process. Please refine your query.");
+        }
+        long totalItems = resultSet.getNumberFound();
+        // Generate a random process id
+        String processId = UUID.randomUUID().toString();
+
+        T initBulkStatus = getInitBulkStatus(processId, totalItems);
+        bulkMonitor.updateBulkStatus(initBulkStatus);
+        bulkMonitor.registerProcess(nodeRef, processId);
+
+        BatchProcessWorker<NodeRef> batchProcessWorker = getWorkerProvider(nodeRef, bulkOperation);
+        BulkStatusUpdater bulkStatusUpdater = getBulkStatusUpdater();
+
+        BatchProcessor<NodeRef> batchProcessor = new BatchProcessor<>(
+            processId,
+            transactionService.getRetryingTransactionHelper(),
+            getWorkProvider(bulkOperation, totalItems, bulkStatusUpdater),
+            threadCount,
+            itemsPerTransaction,
+            bulkStatusUpdater,
+            LOG,
+            loggingInterval);
+
+        runAsyncBatchProcessor(batchProcessor, batchProcessWorker, bulkStatusUpdater);
+        return initBulkStatus;
+    }
+
+    /**
+     * Run batch processor
+     */
+    protected void runAsyncBatchProcessor(BatchProcessor<NodeRef> batchProcessor,
+        BatchProcessWorker<NodeRef> batchProcessWorker, BulkStatusUpdater bulkStatusUpdater)
+    {
+        Runnable backgroundLogic = () -> {
+            try
+            {
+                if (LOG.isDebugEnabled())
+                {
+                    LOG.debug("Started processing batch with name: " + batchProcessor.getProcessName());
+                }
+                batchProcessor.processLong(batchProcessWorker, true);
+                if (LOG.isDebugEnabled())
+                {
+                    LOG.debug("Processing batch with name: " + batchProcessor.getProcessName() + " completed");
+                }
```

Comment:
If we switch to SLF4J these should use appropriate logging format syntax (e.g.: `"Started processing batch with name: {}, batchProcessor.getProcessName()"`)